### ISSUE TITLE
Failures on Test "why not promoted" functionality for extension types.

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/sdk#53741]
There are new test failures on [Test "why not promoted" functionality for extension types.](https://github.com/dart-lang/sdk/commit/cdef17c3cce5953160c899af08e845b99394ee0e).

The tests
```
language/extension_type/why_not_promoted_getter_error_test MissingCompileTimeError (expected Pass)
```
are failing on configurations
```
cfe-strong-linux
cfe-weak-linux
```

**Comment by stereotype441** [comment]
The failure bisects to my CL (cdef17c3cce5953160c899af08e845b99394ee0e), because that is the CL that introduced the failing test.

However, the test passed during presubmit testing (see https://dart-ci.firebaseapp.com/cl/330027/1). Digging a little, I found that if I had landed my CL prior to 7be0e438d993ad47522bcc7700fe4b98377c5fd5, it would have passed.

Looking at the failure details (https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket/8767441706784829937/+/u/test_results/new_test_failures__logs_), it appears that with 7be0e438d993ad47522bcc7700fe4b98377c5fd5, code like this now causes the CFE to produce a compile time error:
```dart
extension type E(dynamic d) {
  int? get i1 => 0;
}

void test(E e) {
  e.i1 == null;
}

main() {}
```
(note, to repro, use `--enable-experiment=inline-class`)
The compile-time error is:
```dart
../../tmp/proj/test.dart:6:5: Error: Property 'i1' cannot be accessed on 'E' because it is potentially null.
Try accessing using ?. instead.
  e.i1 == null;
    ^^
```
I believe there is a flaw in 7be0e438d993ad47522bcc7700fe4b98377c5fd5 causing this bogus error to be emitted.

I've accepted the test failure for now in order to turn the bot green.
